test/test_hpyerr.py

```diff
@@ -341,3 +341,49 @@ def check_warning(cls):
         check_warning(UnicodeWarning)
         check_warning(BytesWarning)
         check_warning(ResourceWarning)
+
+    def test_errorval_returned_by_api_functions_hpy(self):
+        mod = self.make_module("""
+            HPyDef_METH(f, "f", f_impl, HPyFunc_VARARGS)
+            static HPy f_impl(HPyContext ctx, HPy self, HPy *args, HPy_ssize_t nargs)
+            {
+                HPy a = HPy_NULL;
+                HPy b = HPy_NULL;
+                HPy res;
+                if (!HPyArg_Parse(ctx, NULL, args, nargs, "OO", &a, &b))
+                    return HPy_NULL;
+                res = HPy_TrueDivide(ctx, a, b);
+
+                // the point of the test is to check that in case of error
+                // HPy_Div returns HPy_NULL
+                if (HPy_IsNull(res)) {
+                    HPyErr_Clear(ctx);
+                    return HPyLong_FromLong(ctx, -42);
+                }
+
+                HPy_Close(ctx, b);
```

Comment:
`b` should not be closed here.

Reply:
uh, true! We need the debug mode :facepalm:

Reply:
fixed